src/NServiceBus.Core/Routing/FileBasedDynamicRouting/AsyncTimer.cs

```diff
@@ -34,6 +34,10 @@ public void Start(Func<Task> callback, TimeSpan interval, Action<Exception> erro
 
         public Task Stop()
         {
+            if (tokenSource == null)
```

Comment:
The operations below operate on `tokenSource` so I think it would be confusing if the null-check was done on a task.

This is not a FST. The `Start` method is called when first loading the routing table so I think it is better to not make big assumptions here about the start-stop sequence.